### PR TITLE
Re-enable e2e tests with go mod.

### PR DESCRIPTION
**Pull request by n3wscott**
## Proposed Changes

- Moving to go mod broke how eventing is installed from source. Go mod pathing was incorrect. So to move forward with the 300k PR we disabled e2e. This brings them back.

**Release Note**

```release-note
NONE
```

### CODE REVIEW

File: test/config/100-camel-k-1.0.0-RC2.yaml [hunk 1]
```diff
@@ -0,0 +1,447 @@
...
+  name: camel-k-operator-knative
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: camel-k-operator-knative
+subjects:
+- kind: ServiceAccount
+  name: camel-k-operator
+  namespace: camelk
+
```
Comment (mattmoor) [inline test/config/100-camel-k-1.0.0-RC2.yaml:447]: Remove trailing whitespace:
```suggestion

```

Comment (mattmoor) [inline test/config/100-camel-k-1.0.0-RC2.yaml:447]: Add trailing newline:
```suggestion
  

```

**Comment by n3wscott** [comment]
@googlebot come on

File: test/e2e-tests.sh [hunk 2]
```diff
@@ -142,10 +143,10 @@ function test_setup() {
   add_trap "kill $kail_pid || true" EXIT
 
   # Publish test images.
-  echo ">> Publishing test images from vendor"
+  echo ">> Publishing test images from eventing"
   # We vendor test image code from eventing, in order to use ko to resolve them into Docker images, the
   # path has to be a GOPATH.
-  sed -i 's@knative.dev/eventing/test/test_images@knative.dev/eventing-contrib/vendor/knative.dev/eventing/test/test_images@g' "${VENDOR_EVENTING_TEST_IMAGES}"/*/*.yaml
+  sed -i 's@knative.dev/eventing/test/test_images@knative.dev/eventing-contrib/vendor/knative.dev/eventing/test/test_images@g' "${VENDOR_EVENTING_TEST_IMAGES}"*/*.yaml
```
Comment (mattmoor) [inline test/e2e-tests.sh:149]: 🤔 I'm honestly surprised this works.  I couldn't make this work for Serving.

cc @dprotaso

Comment (n3wscott) [reply test/e2e-tests.sh:149]: I had to vendor all the test images.

**Comment by mattmoor** [comment]
Seems to be working, so yolo.

/lgtm

**Pull request merged**